src/header.js

```diff
@@ -50,7 +51,8 @@ class Header extends EthObject {
         toBuffer(rpcResult.timestamp),
         toBuffer(rpcResult.extraData),
         toBuffer(rpcResult.mixHash),
-        toBuffer(rpcResult.nonce)
+        toBuffer(rpcResult.nonce),
+        toBuffer(rpcResult.baseFeePerGas)
```

Comment:
Let's only add this field if it exists, so it works as well with Ethereum.

```javascript
let data = [...];

if (rpcResult.baseFeePerGas !== undefined && rpcResult.baseFeePerGas !== null) {
    data.push(toBuffer(rpcResult.baseFeePerGas));
}

return new this(data);
```

Reply:
`toBuffer(undefined)` is an empty `Uint8Array []` so shouldn't this still serialize correctly for Mainnet also ?

Reply:
I don't think so, because while you are serializing using RLP it checks the number of elements, and and it is different:
`[1, 2, []]`, than `[1, 2]`.